jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/SubnetInventoryCollector.java

```diff
@@ -63,14 +63,23 @@ public List<SubnetVH> fetchSubnetDetails(SubscriptionVH subscription) {
 					if (properties != null) {
 						HashMap<String, Object> propertiesMap = new Gson().fromJson(properties.toString(),
 								HashMap.class);
-						subnetVH.setIpConfigurations((List<Map<String, Object>>) propertiesMap.get("ipConfigurations"));
-						subnetVH.setAddressPrefix(propertiesMap.get("addressPrefix").toString());
-						subnetVH.setPrivateLinkServiceNetworkPolicies(
-								propertiesMap.get("privateLinkServiceNetworkPolicies").toString());
-						subnetVH.setProvisioningState(propertiesMap.get("provisioningState").toString());
-						subnetVH.setPrivateEndpointNetworkPolicies(
-								propertiesMap.get("privateEndpointNetworkPolicies").toString());
-
+						if(propertiesMap.get("ipConfigurations")!=null){
+							subnetVH.setIpConfigurations((List<Map<String, Object>>) propertiesMap.get("ipConfigurations"));
+						}
+						if(propertiesMap.get("addressPrefix")!=null){
+							subnetVH.setAddressPrefix(propertiesMap.get("addressPrefix").toString());
+						}
+						if(propertiesMap.get("privateLinkServiceNetworkPolicies")!=null){
+							subnetVH.setPrivateLinkServiceNetworkPolicies(
+									propertiesMap.get("privateLinkServiceNetworkPolicies").toString());
+						}
+						if(propertiesMap.get("provisioningState")!=null){
+							subnetVH.setProvisioningState(propertiesMap.get("provisioningState").toString());
+						}
+						if(propertiesMap.get("provisioningState")!=null){
+							subnetVH.setPrivateEndpointNetworkPolicies(
+									propertiesMap.get("privateEndpointNetworkPolicies").toString());
+						}
```

Comment:
Pls reformat your code, basic InteliJ reformat is pretty much enough (Ctrl+Alt+L). If it's not available - use line 63 as an example.

Reply:
Intead of `.get("asdfasdf") != null` use method `has` - its purpose is exactly what you need.

Reply:
Reformatted the code. 
Regarding second point, what if map has the key but has null as value.  map.containsKey("asdfasdf") return true but value may be still null, again we need an additional check. To be on safer side, lets explicitly check for null value.

Reply:
oh, yup, missed it, you are right.